pkg/sources/adapter/azureservicebussource/adapter.go

```diff
@@ -230,22 +248,49 @@ func connectionStringFromEnvironment(namespace, entityPath string) string {
 //  Both (DataAction):
 //  - Microsoft.ServiceBus/namespaces/messages/receive/action
 func (a *adapter) Start(ctx context.Context) error {
+	maxMessages := 5
 	logging.FromContext(ctx).Info("Listening for messages")
 
-	handle := a.msgRcvr.Listen(ctx, servicebus.HandlerFunc(a.handleMessage))
-	<-handle.Done()
-	return handle.Err()
+	messages, err := a.msgRcvr.ReceiveMessages(ctx, maxMessages, nil)
+	if err != nil {
+		return fmt.Errorf("error receiving messages: %w", err)
+	}
+
+	for _, m := range messages {
+		body, err := m.Body()
+		if err != nil {
+			return fmt.Errorf("error getting message data: %w", err)
+		}
+
+		msg := &Message{
+			Data:            body,
+			ReceivedMessage: *m,
+			LockToken:       stringifyLockToken((*uuid.UUID)(&m.LockToken)),
+		}
+
+		err = a.handleMessage(ctx, msg)
```

Comment:
If I add the `azservicebus.ReceviedMessage` in the func `handleMessage `we won't be able to have a test for it, because the `azservicebus.ReceviedMessage` doesn't have the `Data` parameter, to get it, we have to call the msg.Body() func

Reply:
OK fair enough, I can't think of a better way right now.